pyinfra/connectors/ssh.py

```diff
@@ -227,6 +229,14 @@ def connect(state: "State", host: "Host"):
     design so can be run in parallel.
     """
 
+    retries = state.config.CONNECT_RETRIES
```

Comment:
I think this should be defined as host data (see the `DATA_KEYS` in this file) rather than config, that gives us more flexibility to control things per host.

So I'd have like defaults as so:

```
host.data.ssh_connect_retries = 0
host.data.ssh_connect_retry_min_delay = 0.1
host.data.ssh_connect_retry_max_delay = 0.5
```

Reply:
Sure, I moved it it to host data and added a unit-test, which fixed also a bug (range is excluding stop).
```diff
-  for tries_left in range(retries, 0, -1):
+  for tries_left in range(retries, -1, -1): 
```